src/pybind/mgr/nfs/export.py

```diff
@@ -637,6 +640,13 @@ def create_cephfs_export(self,
                              access_type: str,
                              clients: list = [],
                              sectype: Optional[List[str]] = None) -> Tuple[int, str, str]:
+        fs_client = CephfsClient(self.mgr)
```

Comment:
This would create a cephfs client on each export create request.

Reply:
Hmm right, I'll find some other way to get the fs, thanks for pointing this out

Reply:
@vshankar https://github.com/ceph/ceph/pull/49460/commits/6c1e06e6d888c354f0e0f821dff594bbd2d16806, this would prevent `__init__` execution and would only give the `fs`. PTAL.

Reply:
this would not work as well, looking for something else

Reply:
Have you looked at how its done in mgr/volumes?

Reply:
should i derive a class here from `CephfsClient` and override it's `__init__`?

Reply:
> Have you looked at how its done in mgr/volumes?


In `mgr/volumes`, only `volume.py` and `async_job.py` use `CephfsClient`, where `async_job.py`'s case is related (cause each async job group uses its own libcephfs connection), but in `volume.py`, class `VolumeClient` is created whose parent is `CephfsClient` and some overriding happens there.

Reply:
> should i derive a class here from `CephfsClient` and override it's `__init__`?

Right. That way you can use the open_filesystem context manager helper. That way you need not worry about cleaning up the connections - the fs handles get garbage collected eventually.

Reply:
I was thinking maybe we can do something like this in `export.py` as we have `mgr` there:

```
fs = None
        fs_map = self.mgr.get('fs_map')
        for filesystem in fs_map['filesystems']:
            if filesystem['mdsmap']['fs_name'] == fs_name:
                fs = filesystem
```
All we need is fs so that we can do statx on it. Makes sense?

Reply:
> I was thinking maybe we can do something like this in `export.py` as we have `mgr` there:
> 
> ```
> fs = None
>         fs_map = self.mgr.get('fs_map')
>         for filesystem in fs_map['filesystems']:
>             if filesystem['mdsmap']['fs_name'] == fs_name:
>                 fs = filesystem
> ```
> 
> All we need is fs so that we can do statx on it. Makes sense?

`filesystem` is _not_ the cephfs connection handle here. How would you do stat on that?